product-service/getProductById.ts

```diff
@@ -0,0 +1,40 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import 'source-map-support/register';
+
+import products from './db/productList.json';
+
+export const handler: APIGatewayProxyHandler = async (event) => {
+  const id = event.pathParameters?.id;
```

Comment:
You are not logging anything, what would be in case of support, API is not working and you have to understand why based on CW logs.